### PR TITLE
Deserializing additional data throughout the object hierarchy

### PR DESCRIPTION
**Pull request by dadheech115**
Fixes #
Fix for - AdditionalDataManager was not getting populated for child objects while deserializing the json

### Changes proposed in this pull request
Have added a function in DefaultSerializer class which gets called recursively to set additional data on Child objects as well

### CODE REVIEW

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java [hunk 1]
```diff
@@ -100,13 +100,59 @@ public DefaultSerializer(final ILogger logger) {
...
     }
 
+    /**
+     * Recursively sets additional data for each child object
+     *
+     * @param serializedObject   the parent object whose children will be iterated to set additional data
+     * @param rawJson            the raw json
+     */
+    @SuppressWarnings("unchecked")
+    private void setChildAdditionalData(IJsonBackedObject serializedObject, JsonObject rawJson) {
```
Comment (davidmoten) [inline src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: No unit tests for this method. Can they be added?

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java [hunk 2]
```diff
@@ -100,13 +100,59 @@ public DefaultSerializer(final ILogger logger) {
...
+                // If the object is a valid Graph object, set its additional data
+                else if (fieldObject != null && fieldObject instanceof IJsonBackedObject) {
+                    IJsonBackedObject serializedChild = (IJsonBackedObject) fieldObject;
+                    AdditionalDataManager childAdditionalDataManager = serializedChild.additionalDataManager();
+
+                    childAdditionalDataManager.setAdditionalData(rawJson.get(field.getName()).getAsJsonObject());
+                    setChildAdditionalData((IJsonBackedObject) fieldObject,rawJson.get(field.getName()).getAsJsonObject());
+                }
+            } catch (IllegalArgumentException | IllegalAccessException e) {
+                logger.logError("Unable to access child fields of " + serializedObject.getClass().getSimpleName(), e);
```
Comment (davidmoten) [inline src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: What is the argument for not throwing? Can comments be added here to that effect?

**Comment by davidmoten** [comment]
@dadheech115 can you add unit tests for this PR?

**Comment by davidmoten** [comment]
@deepak2016 Is @dadheech115 around? Can I get a response?

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java (hunk 1 above)
Comment (dadheech115) [reply src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: Thanks for bringing it up. I will add the Unit test.

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java (hunk 2 above)
Comment (dadheech115) [reply src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: I will add the comments for more clarification.

**Comment by davidmoten** [comment]
@dadheech115 or @deepak2016 
Please edit the initial comment on this PR (not the title!) so that it says "Fixes #104". At the moment it says "Fixes #" only.

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java (hunk 2 above)
Comment (dadheech115) [reply src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: I have raised a separate PR for this - #114 .

File: src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java (hunk 1 above)
Comment (dadheech115) [reply src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java]: I have raised a separate PR for this - #114 .

**Comment by davidmoten** [comment]
@dadheech115 or @deepak2016 
Please fix this PR as requested above. I am asking you to edit the comments, that's all!

**Comment by dadheech115** [comment]
@davidmoten , I have added the issue number.

**Pull request merged**